addons/l10n_in_ewaybill_stock_purchase/models/stock_move.py

```diff
@@ -0,0 +1,17 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _compute_ewaybill_price_unit(self):
+        super()._compute_ewaybill_price_unit()
+        for line in self.filtered(lambda l: l.purchase_line_id):
+            line.ewaybill_price_unit = line.purchase_line_id.price_unit
```

Comment:
this could be fetched from the stock valuation layers

Reply:
Why?? We prefer `price_unit` from `purchase_line_id` it's more straight, simple and accurate

Reply:
@hamo-odoo But then you have to depend on purchase and that way all the information is already from the picking.  Also, the stock valuation could work for more the cases at once.  (purchase, sale (not sure because that would be at cost))  It would be interesting however to know why you think it would be less accurate for purchases.

Reply:
@jco-odoo What about `consumable` products??

Reply:
@hamo-odoo Good point.